Mathlib/Data/Nat/Bitset.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2024 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Combinatorics.Colex
+
+/-!
+# Bitset
+
+Given `n : ℕ`, we define `Nat.bitSet n`, which is the `Finset` of indices that have a `1` in the
+binary expansion of `n`. We show that this function is the inverse of `fun s ↦ ∑ i in s, 2^i`,
+and therefore that it gives an order isomorphism from `ℕ` to `Colex ℕ`.
+
+The latter proof uses the machinery in `Combinatorics.Colex`, which is why this doesn't appear
+in an earlier file. The implementation of `Nat.bitSet` is a direct recursive definition,
+which seems cleaner than the other option of using `Nat.bits` and therefore the `List` API.
+
+TODO : Relate the material in this file to `Nat.digits` and `Nat.bits`.
+-/
+
+open Finset BigOperators
+
+namespace Nat
+
+variable {a n : ℕ} {s : Finset ℕ}
+
+/-- The function which maps the natural number `∑ i in s, 2^i` to the Finset `s`.
+This could also be defined using `Nat.bits`, but it seems easier to avoid the `List` api.  -/
+def bitSet (n : ℕ) : Finset ℕ := by
+  induction' n using binaryRec with b _ s
+  · exact ∅
+  cases b
+  · exact image (· + 1) s
```

Comment:
Can you use `map` here instead? It will be faster and have nicer defeqs. Same for cons instead of insert below.

Reply:
Ok. I think `indexesOf` rather than `map` is the right thing though.